raiden/exceptions.py

```diff
@@ -279,6 +283,10 @@ class BrokenPreconditionError(RaidenError):
     """
 
 
+class BrokenPreconditionDuplicatedChannelError(BrokenPreconditionError, DuplicatedChannelError):
```

Comment:
Please remove this new exception and use `BrokenPreconditionError`. The point is, we do not want to introduce specific `BrokenPreconditionXError` for each of the errors knowing that ALL precondition errors are just going to crash the client for "unexpected" behaviour.

Reply:
What now, does my fix "generally look good" or am I supposed to throw it away? The reason for the test failures is that `ConnectionManager._join_partner` relied on the contract proxy raising a `DuplicatedChannelError` to detect already existing channels. With the recent changes to the proxy this is no longer true, it may raise anything now depending on when the channel creation happened:

- a `BrokenPreconditionError` if its visible in the precondition checks already,
- a `DuplicatedChannelError` if it happens to be not yet visible in the precondition checks but already onchain
- a `RaidenRecoverableError` (gas estimation failed for unknown reason) if the channel is only visible in the pending block. 

My fix is:
- catch the relevant cases of `BrokenPreconditionError` too. That's why the subclasses are introduced.
- handle the `RaidenRecoverableError`s by checking for a new channel again. Of course, the contract proxy should handle this case itself and never raise a `RaidenRecoverableError`, but I've given up on arguing about the contract proxies at this point.

Reply:
> catch the relevant cases of BrokenPreconditionError too. That's why the subclasses are introduced.

Please, do not catch `PreconditionError`s, instead do the check for the trigerring block before calling the `new_netting_channel`

Reply:
Hmm, I just realize that `RaidenAPI.channel_open` already checks that and the `BrokenPreconditionError` should not happen. I'll try to fix it there without meddling with the exception classes.

(The problem with the inconsistent handling of failed gas estimations remains though, the workaround for that will still be needed.)

Reply:
> What now, does my fix "generally look good" or am I supposed to throw it away?

I don't follow, but i take that you're not happy with the comment asking you to remove stuff although i commented on why i was asking for the change?

Reply:
Sorry for this exagerrated reply. You were right to request that change, but basically it made me revert my whole fix and write a new one. That is what I meant, the entire original PR was based on subclassing and catching the `BrokenPreconditionError`.